tests/test_datastore.py

```diff
@@ -1951,5 +1951,18 @@ def test_StoredFileInfo(self) -> None:
         self.assertEqual(unpickled_info, info)
 
 
+@contextlib.contextmanager
+def _temp_yaml_file(data: Any) -> Iterator[str]:
+    fh = tempfile.NamedTemporaryFile(mode="w", suffix=".yaml")
+    try:
+        yaml.dump(data, stream=fh)
+        fh.flush()
+        yield fh.name
+    finally:
+        # Some tests delete the file
```

Comment:
Yes, the tests are wanting to make sure that tests that said the file was going to be deleted really did delete the file. `lsst.utils.tests.getTempFilePath` does do that test. Can you add a parameter for that here as well so that we can test move?

Reply:
I added the check back, but I put it in testIngestTransfer.succeed where it's more local to the code creating the situation being tested.